diff --git a/xr_usb_serial_common-1a/xr_usb_serial_common.c b/xr_usb_serial_common-1a/xr_usb_serial_common.c
--- a/xr_usb_serial_common-1a/xr_usb_serial_common.c
+++ b/xr_usb_serial_common-1a/xr_usb_serial_common.c
@@ -916,8 +916,13 @@ static int xr_usb_serial_tty_ioctl(struct tty_struct *tty,
 	return rv;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+static void xr_usb_serial_tty_set_termios(struct tty_struct *tty,
+						const struct ktermios *termios_old)
+#else
 static void xr_usb_serial_tty_set_termios(struct tty_struct *tty,
 						struct ktermios *termios_old)
+#endif
 {
 	struct xr_usb_serial *xr_usb_serial = tty->driver_data;
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 7, 0)
